### PR TITLE
including release in dataset

### PR DESCRIPTION
**Pull request by Nathanjp91**
Added release to the dataset object. Using get_remote_dataset with a string containing a release (ie 'v7-labs/random-image:testexport1')will set the release and download that rather than latest.

### CODE REVIEW

**Review by owencjones — COMMENTED** [review]
Any way we can have a unit test that prevents this confusion of terminology happening in the same part of the code again?

**Comment by Nathanjp91** [comment]
> Any way we can have a unit test that prevents this confusion of terminology happening in the same part of the code again?

Added some tests to make sure the slugs/id were set correctly in the database object but given most of the functionality is around pulling data from the server, it would be very difficult for full coverage.

File: tests/darwin/dataset/remote_dataset_test.py [hunk 1]
```diff
@@ -306,6 +306,24 @@ def files_content() -> Dict[str, Any]:
     }
 
 
+def test_dataset_creation_from_id():
```
Comment (owencjones) [inline tests/darwin/dataset/remote_dataset_test.py:309]: Implementation tests only really, but I appreciate the reasoning.

**Review by owencjones — APPROVED** [review]
LGTM

**Pull request merged**